### ISSUE TITLE
Ratio 精度误差讨论

### ISSUE THREAD

**Issue by Jiexuanz** [XueZeyue/DanceGRPO#57]
flow-grpo 那边作者说明：一些 setting 的设置是因为处理 ratio 精度问题，不这样设置在训练时 ratio 会出现精度的 diff，咱们这边代码有这样的问题吗

flow-grpo 的 ratio diff 还蛮明显的，我在实际测试的时候

**Comment by XueZeyue** [comment]
https://github.com/XueZeyue/DanceGRPO/blob/main/README.md?plain=1#L108

I set the rollout inference batch size and train batch size to one; otherwise, the ratio will not be one.

**Comment by XueZeyue** [comment]
> Thanks for the detailed explanation! In SD1.4 with bf16 mixed training, I found sometimes the ratio (the last timestep to be specific) can be really off e.g., 0.98xx. Does that make sense? Other than the last timestep, it's usually normal, e.g. 0.999x to 1.000x.
> 
> Also, may I ask why setting the batch size to 1 would solve this problem? Is it due to batch normalization or some other operations? Thanks!

Hi, I also don't understand why I have to set the batch size to 1, but only when I set the batch size to 1, the results in FastVideo and the internal model at Seed will have a ratio of 1.0. For DiT models, we use LN rather than BN.
